presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -633,6 +633,13 @@ folly::Future<std::unique_ptr<Result>> TaskManager::getResults(
 
     for (;;) {
       if (prestoTask->taskStarted) {
+        // If the task is failed, then return an error. Before accessing velox
+        // task state, we need to test this as velox task can be null if there
+        // is an error present.
+        if (prestoTask->error != nullptr) {
+          VELOX_USER_FAIL("Calling getResult() on a failed task: {}", taskId);
+        }
```

Comment:
@amitkdutta CAn you please doublecheck that it is impossible to have this situation:
1. getData() request comes and either it creates the new PrestoTask or some other thread does.
2. That new PrestoTask does not have Velox Task (it is null).
3. We throw this error.
4. Velox Task is created for this PrestoTask.

Reply:
There is this special scenario that creates an empty PrestoTask without Velox Task in it. This scenario happens when error occurs while parsing or doing any protocol translation in task update endpoint. The empty shell PrestoTask was created to contain that error so that following getInfo()/getStatus() calls on that task can know what's going on. I think the crash here might be related to taht.

Reply:
A way to verify is to print out the contained error to see if it is an error related to parsing/json/protocol etc.

Reply:
> So, this fix will not necessarily will fix the problem. By that time we might still not have error in the task. @tanjialiang Can the task in this special case be created with error from the start to avoid the race?

The task(PrestoTask) is created from the start with Error in it. We cannot create a Velox task because error happens while creating that velox task.